rclone-mixin/rclone.libsonnet

```diff
@@ -0,0 +1,193 @@
+                             fields: '',
+                           },
+                           pieType: 'donut',
+                           tooltip: {
+                             mode: 'single',
+                           },
+                           legend: {
+                             displayMode: 'table',
+                             placement: 'right',
+                             values: [
+                               'value',
+                               'percent',
+                             ],
+                           },
+                         },
+                       };
+
+local transferGauge = singlestat.new(
+  'Transfer Rate',
+  description='Thresholds are; 0-80MB/s (Internet speed), 80-200MB/s (Harddisk speed), 200MB/s+ (Solid State disk speed)',
+  datasource='$datasource',
+  gaugeShow=true,
+  format='Bps',
+  span=2,
+  thresholds='80000000,200000000',
+  gaugeMaxValue=500000000
+)
+                      .addTarget(grafana.prometheus.target(queries.bytes_transferred_rate_sum));
+
+local transferredStat = statPanel.new(
+  'Data Transferred',
+  datasource='$datasource',
+  unit='bytes',
+)
+                        .addTargets([
+  grafana.prometheus.target(queries.bytes_transferred_range, legendFormat='Range'),
+  grafana.prometheus.target(queries.bytes_transferred_day, legendFormat='24h'),
+  grafana.prometheus.target(queries.bytes_transferred_total, legendFormat='Total'),
+])
+                        .addThreshold({ color: 'green', value: 0 }) { span: 5 };
+
+local transferGraph = graphPanel.new(
+                        'Traffic Rate',
+                        datasource='$datasource',
+                        span=6,
+                        stack=true,
+                        format='bps',
+                      )
+                      .addTarget(grafana.prometheus.target(queries.bytes_transferred_rate_ts, legendFormat='{{instance}}'))
+                      + {
+                        line: 1,
+                        fill: 5,
+                        fillGradient: 10,
+                      };
+
+local statusGraph = graphPanel.new(
+  'State Rates',
+  datasource='$datasource',
+  span=6,
+)
+                    .addTargets([
+  grafana.prometheus.target(queries.fchecked_irate, legendFormat='Files Checked'),
+  grafana.prometheus.target(queries.ftransferred_irate, legendFormat='Files Transferred'),
+  grafana.prometheus.target(queries.frenamed_irate, legendFormat='Files Renamed'),
+  grafana.prometheus.target(queries.fdeleted_irate, legendFormat='Files Deleted'),
+  grafana.prometheus.target(queries.fdir_deleted_irate, legendFormat='Directories Deleted'),
+  grafana.prometheus.target(queries.err_irate, legendFormat='Transfer Errors'),
+]);
+
+local jobsTable = jt.new('$datasource', dashboardUid)
+                  .addTargets([
+  grafana.prometheus.target(queries.starttime, format='table'),
+  grafana.prometheus.target(queries.endtime, format='table'),
+]);
```

Comment:
Here is the suggestion to format indentation and make it a bit more clean. If you move RHS after newline then methods or patches would have neat indentation after ```jsonnetfmt -n 2 --max-blank-lines 1 --string-style s --comment-style s``` applied.

```suggestion
local job_template =
  grafana.template.new(
    'job',
    '$datasource',
    'label_values(rclone_speed, job)',
    label='job',
    refresh='load',
    multi=true,
    includeAll=true,
    allValues='.+',
    sort=1,
  );

local instance_template =
  grafana.template.new(
    'instance',
    '$datasource',
    'label_values(rclone_speed{job=~"$job"}, instance)',
    refresh='load',
    multi=true,
    includeAll=true,
    allValues='.+',
    sort=1,
  );

local statusPiePanel =
  piechart.new('Status', span=5, datasource='$datasource', legendType='Right side')
  .addTarget(grafana.prometheus.target(queries.fchecked_total, legendFormat='Files Checked'))
  .addTarget(grafana.prometheus.target(queries.ftransferred_total, legendFormat='Files Transferred'))
  .addTarget(grafana.prometheus.target(queries.frenamed_total, legendFormat='Files Renamed'))
  .addTarget(grafana.prometheus.target(queries.fdeleted_total, legendFormat='Files Deleted'))
  .addTarget(grafana.prometheus.target(queries.fdir_deleted_total, legendFormat='Directories Deleted'))
  .addTarget(grafana.prometheus.target(queries.err_total, legendFormat='Transfer Errors'))
  + {
    type: 'piechart',
    options: {
      reduceOptions: {
        values: false,
        calcs: [
          'lastNotNull',
        ],
        fields: '',
      },
      pieType: 'donut',
      tooltip: {
        mode: 'single',
      },
      legend: {
        displayMode: 'table',
        placement: 'right',
        values: [
          'value',
          'percent',
        ],
      },
    },
  };

local transferGauge =
  singlestat.new(
    'Transfer Rate',
    description='Thresholds are; 0-80MB/s (Internet speed), 80-200MB/s (Harddisk speed), 200MB/s+ (Solid State disk speed)',
    datasource='$datasource',
    gaugeShow=true,
    format='Bps',
    span=2,
    thresholds='80000000,200000000',
    gaugeMaxValue=500000000
  )
  .addTarget(grafana.prometheus.target(queries.bytes_transferred_rate_sum));

local transferredStat =
  statPanel.new(
    'Data Transferred',
    datasource='$datasource',
    unit='bytes',
  )
  .addTargets([
    grafana.prometheus.target(queries.bytes_transferred_range, legendFormat='Range'),
    grafana.prometheus.target(queries.bytes_transferred_day, legendFormat='24h'),
    grafana.prometheus.target(queries.bytes_transferred_total, legendFormat='Total'),
  ])
  .addThreshold({ color: 'green', value: 0 }) { span: 5 };

local transferGraph =
  graphPanel.new(
    'Traffic Rate',
    datasource='$datasource',
    span=6,
    stack=true,
    format='bps',
  )
  .addTarget(grafana.prometheus.target(queries.bytes_transferred_rate_ts, legendFormat='{{instance}}'))
  + {
    line: 1,
    fill: 5,
    fillGradient: 10,
  };

local statusGraph =
  graphPanel.new(
    'State Rates',
    datasource='$datasource',
    span=6,
  )
  .addTargets([
    grafana.prometheus.target(queries.fchecked_irate, legendFormat='Files Checked'),
    grafana.prometheus.target(queries.ftransferred_irate, legendFormat='Files Transferred'),
    grafana.prometheus.target(queries.frenamed_irate, legendFormat='Files Renamed'),
    grafana.prometheus.target(queries.fdeleted_irate, legendFormat='Files Deleted'),
    grafana.prometheus.target(queries.fdir_deleted_irate, legendFormat='Directories Deleted'),
    grafana.prometheus.target(queries.err_irate, legendFormat='Transfer Errors'),
  ]);

local jobsTable =
  jt.new('$datasource', dashboardUid)
  .addTargets([
    grafana.prometheus.target(queries.starttime, format='table'),
    grafana.prometheus.target(queries.endtime, format='table'),
  ]);
```